api/src/pcapi/core/token/__init__.py

```diff
@@ -0,0 +1,87 @@
+import dataclasses
+from datetime import datetime
+from datetime import timedelta
+import enum
+from typing import Type
+from typing import TypeVar
+
+from flask import current_app as app
+import jwt
+
+from pcapi.core.users import utils
+from pcapi.core.users.exceptions import InvalidToken
+
+
+class TokenType(enum.Enum):
+    EMAIL_CHANGE_CONFIRMATION = "update_email_confirmation"
+    EMAIL_CHANGE_VALIDATION = "update_email_validation"
+
+
+@dataclasses.dataclass(frozen=True)
+class Token:
+    type_: TokenType
+    user_id: int
+    encoded_token: str
+    data: dict
+
+    @classmethod
+    def load(cls: "Token", encoded_token: str) -> "Token":
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🤓 nitpick:** ‏J'avais pas fais attenmtion quand on en parlait tout à l'heure, pour une `classmethod` pas besoin de typer explicitement le param `cls`, mypy sait que c'est du type de la classe